Essay_Analysis_Tool/DiffMergeForm.cs

```diff
@@ -0,0 +1,585 @@
+    {
+        private int _updating;
+        private static readonly Style _greenStyle;
+        private static readonly Style _redStyle;
+
+        static DiffMergeForm()
+        {
+            _greenStyle = new MarkerStyle(new SolidBrush(Color.FromArgb(50, Color.Lime)));
+            _redStyle = new MarkerStyle(new SolidBrush(Color.FromArgb(50, Color.Red)));
+        }
+
+        public DiffMergeForm()
+            : this("") { }
+
+        /// <summary>
+        /// Initialize a new form.
+        /// </summary>
+        /// <param name="file">Absolute or relative file path.</param>
+        public DiffMergeForm(string file)
+            : this(file, "") { }
+
+        /// <summary>
+        /// Initialize a new form.
+        /// </summary>
+        /// <param name="file1">Absolute or relative file path.</param>
+        /// <param name="file2">Absolute or relative file path.</param>
+        public DiffMergeForm(string file1, string file2)
+        {
+            InitializeComponent();
+
+            tbFirstFile.Text = file1;
+            tbSecondFile.Text = file2;
+        }
+
+        private void SelectFirstFileButton_Click(object sender, EventArgs e)
+        {
+            if (ofdFile.ShowDialog() == DialogResult.OK)
+                tbFirstFile.Text = ofdFile.FileName;
+        }
+
+        private void SelectSecondFileButton_Click(object sender, EventArgs e)
+        {
+            if (ofdFile.ShowDialog() == DialogResult.OK)
+                tbSecondFile.Text = ofdFile.FileName;
+        }
+
+        private void CompareButton_Click(object sender, EventArgs e)
+        {
+            if (!File.Exists(tbFirstFile.Text) && !File.Exists(tbSecondFile.Text))
+            {
+                MessageBox.Show(this, "Please select a valid file", "Invalid file");
+                return;
+            }
+
+            fctb1.Clear();
+            fctb2.Clear();
+
+            Cursor = Cursors.WaitCursor;
+
+            if (Path.GetExtension(tbFirstFile.Text).ToLower() == ".cs")
+                fctb1.Language = fctb2.Language = Language.CSharp;
+            else
+                fctb1.Language = fctb2.Language = Language.Custom;
+
+            var source1 = Lines.Load(tbFirstFile.Text);
+            var source2 = Lines.Load(tbSecondFile.Text);
+
+            source1.Merge(source2);
+
+            BeginUpdate();
+
+            Process(source1);
+
+            EndUpdate();
+
+            Cursor = Cursors.Default;
+        }
+
+        private void FastColoredTextBox_VisibleRangeChanged(object sender, EventArgs e)
+        {
+            if (_updating > 0)
+                return;
+
+            var vPos = (sender as FastColoredTextBox).VerticalScroll.Value;
+            var curLine = (sender as FastColoredTextBox).Selection.Start.iLine;
+
+            if (sender == fctb2)
+                UpdateScroll(fctb1, vPos, curLine);
+            else
+                UpdateScroll(fctb2, vPos, curLine);
+
+            fctb1.Refresh();
```

Comment:
If i remove the two `fctb.Refresh()` method the line number section become less laggy but it updates the two FCTB instance less often so when scrolling the text content jump instead of smoothly move left/right/top/down in the screen.